src/index.ts

```diff
@@ -51,6 +59,11 @@ export interface OAuthInterceptorConfig {
    */
   key: string;
 
+  /**
+   * OAuth realm
+   */
+  realm: string | undefined;
```

Comment:
I think you have to use `?:` so that the field is optional.  Otherwise it might be required anyway, but you would have to set it to `undefined`

```suggestion
  realm?: string | null;
```